.drone.yml

```
@@ -1,453 +1,660 @@
-    pull: true
-    environment:
-      TAGS: bindata sqlite sqlite_unlock_notify
-    commands:
-      - make clean
-      - make generate
-      - make vet
-      - make lint
-      - make fmt-check
-      - make swagger-check
-      - make swagger-validate
-      - make misspell-check
-      - make test-vendor
-      - make build
-    when:
-      event: [ push, tag, pull_request ]
-
-  unit-test:
-    image: golang:1.12
-    pull: true
-    group: test
-    environment:
-      TAGS: bindata sqlite sqlite_unlock_notify
-    commands:
-      - make unit-test-coverage
-    when:
-      event: [ push, pull_request ]
-      branch: [ master ]
-
-  release-test:
-    image: golang:1.12
-    pull: true
+    remote: "git@github.com:go-gitea/gitea.git"
+  environment:
+    GIT_PUSH_SSH_KEY:
+      from_secret: git_push_ssh_key
+
+- name: upload_translations
+  pull: always
+  image: jonasfranz/crowdin
+  settings:
+    files:
+      locale_en-US.ini: options/locale/locale_en-US.ini
+    ignore_branch: true
+    project_identifier: gitea
+  environment:
+    CROWDIN_KEY:
+      from_secret: crowdin_key
+
+trigger:
+  branch:
+  - master
+  event:
+  - push
+
+---
+kind: pipeline
+name: default    
+
+platform:
+  os: linux
+  arch: amd64
+
+workspace:
+  base: /go
+  path: src/code.gitea.io/gitea
+
+steps:
+- name: pre-build
+  pull: always
+  image: webhippie/nodejs:latest
+  commands:
+  - npm install
+  - make stylesheets-check
+
+- name: build-without-gcc
+  pull: always
+  image: golang:1.10
+  commands:
+  - go build -o gitea_no_gcc
+
+- name: build
+  pull: always
+  image: golang:1.12
+  commands:
+  - make clean
+  - make generate
+  - make vet
+  - make lint
+  - make fmt-check
+  - make swagger-check
+  - make swagger-validate
+  - make misspell-check
+  - make test-vendor
+  - make build
+  environment:
+    TAGS: bindata sqlite sqlite_unlock_notify
+
+- name: release-test
+  pull: always
+  image: golang:1.12
+  commands:
+  - make test
+  settings:
     group: test
-    environment:
-      TAGS: bindata sqlite sqlite_unlock_notify
-    commands:
-      - make test
-    when:
-      event: [ push, pull_request ]
-      branch: [ release/* ]
-
-  tag-test:
-    image: golang:1.12
-    pull: true
+  environment:
+    TAGS: bindata sqlite sqlite_unlock_notify
+  when:
+    branch:
+    - "release/*"
+    event:
+    - push
+    - pull_request
+
+- name: tag-test
+  pull: always
+  image: golang:1.12
+  commands:
+  - make test
+  settings:
     group: test
-    environment:
-      TAGS: bindata
-    commands:
-      - make test
-    when:
-      event: [ tag ]
-
-  test-sqlite:
-    image: golang:1.12
-    pull: true
+  environment:
+    TAGS: bindata
+  when:
+    event:
+    - tag
+
+---
+kind: pipeline
+name: test-sqlite
+
+workspace:
+  base: /go
+  path: src/code.gitea.io/gitea
+
+steps:
+- name: build
+  pull: always
+  image: golang:1.12
+  commands:
+  - go build -mod=vendor -tags 'sqlite sqlite_unlock_notify' -o gitea
```

shouldn't this be ? (same for other test build):
```suggestion
  - make generate
  - make build
```

---

go generate doesn't have `-mod=vendor` flag, and so it'll fetch all the packages (which adds time to build). So that's why I'm using direct commands.

---

```
go generate code.gitea.io/gitea/modules/public code.gitea.io/gitea/modules/templates
go build -mod=vendor -tags 'bindata sqlite sqlite_unlock_notify' -o gitea
```

---

@lunny even with those changes it will still try to fetch go modules

---

Is it go generate that's getting the modules or one of the things called by go generate?

Eg. Is the go fmt command in the generate header responsible for this?

---

https://github.com/go-gitea/gitea/blob/master/modules/public/public.go#L21